src/ParallelAlgorithms/SurfaceFinder/Python/FindRadialSurface.py

```diff
@@ -0,0 +1,274 @@
+            block_neighbors = [
+                block_neighbor_id
+                for direction, block_neighbor_id in domain.blocks[
+                    block_id
+                ].neighbors.items()
+                if direction.dimension == 2
+                and domain.blocks[block_neighbor_id].has_distorted_frame()
+            ]
+            valid_blocks = [block_id, *block_neighbors]
+            if element.id.block_id in valid_blocks:
+                continue
+            angular_block_logical = np.array(block_logical_coord.data)[:2]
+            angular_element_id = ElementId[2](
+                element.id.block_id, element.id.segment_ids[:2]
+            )
+            angular_element_logical = element_logical_coordinates(
+                angular_block_logical, angular_element_id
+            )
+            if angular_element_logical is None:
+                continue
+            angular_coords.append(angular_element_logical)
+            offsets.append(i)
+        if len(angular_coords) == 0:
+            continue
+        # Find the surface in this element along each ray
+        angular_coords = np.array(angular_coords).T
+        surface_logical_radii = spectre.SurfaceFinder.find_radial_surface(
+            data[0], target, element.mesh, angular_coords
+        )
+        # For those rays where we found the surface in this element, map the
+        # surface radius to inertial coordinates
+        for offset, surface_logical_radius, angular_element_logical in zip(
+            offsets, surface_logical_radii, angular_coords.T
+        ):
+            if surface_logical_radius is None:
+                continue
+            surface_logical_point = [
+                *angular_element_logical,
+                surface_logical_radius,
+            ]
+            surface_radii[offset] = np.linalg.norm(
+                np.array(
+                    element.map(
+                        np.array(surface_logical_point).reshape((3, 1)),
+                        time,
+                        functions_of_time,
+                    )
+                )[:, 0]
+            )
+            filled[offset] = True
+        if np.all(filled):
+            break
+    if not np.all(filled):
+        raise ValueError("Unable to find the surface.")
+    surface = Strahlkorper(
+        l_max=initial_guess.l_max,
+        m_max=initial_guess.m_max,
+        radius_at_collocation_points=surface_radii,
+        center=initial_guess.expansion_center,
+    )
+    logger.info(
+        f"Found radial surface where {var_name} = {target:g} with average"
+        f" radius {surface.average_radius:g} and center"
+        f" {surface.expansion_center}."
+    )
+    # Write the horizon to a file and return it
+    if output_surfaces_file:
+        assert output_coeffs_subfile or output_coords_subfile, (
+            "Specify either 'output_coeffs_subfile' or 'output_coords_subfile'"
+            " or both."
+        )
```

Comment:
Can you move this check to the beginning of the function so that you can learn this info before we do a bunch of work trying to find the surface? I would be a bit annoyed if I had to wait to find a surface (I don't know how long it'll take) and then got this...